FakebookNotifications/FakebookNotifications.Testing/IntegrationTests/NotificationControllerTest.cs

```diff
@@ -0,0 +1,125 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace FakebookNotifications.Testing.IntegrationTests
+{
+    public class NotificationControllerTest
+    {
+
+        private readonly Domain.Models.User _testUser1 = new()
+        {
+            Id = "01",
+            Connections = new List<string>{
+                "00", "02", "03"
+            },
+            Email = "test@test.com"
+        };
+
+        private readonly Mock<IGroupManager> _mockGroups = new();
+        private readonly Mock<IClientProxy> _mockClientProxy = new();
+        private readonly Mock<IUserRepo> _userRepo = new();
+        private readonly Mock<INotificationsRepo> _noteRepo = new();
+        private readonly Mock<IHubContext<NotificationHub>> _mockHubContext = new();
+
+        public NotificationControllerTest()
+        {
+            _mockHubContext.Setup(t => t.Clients.All).Returns(_mockClientProxy.Object);
+            _mockHubContext.Setup(t => t.Groups).Returns(_mockGroups.Object).Verifiable();
+            _mockHubContext.Setup(t => t.Clients.Group(_testUser1.Email)).Returns(_mockClientProxy.Object);
+            _noteRepo.Setup(t => t.CreateNotificationAsync(It.IsAny<Domain.Models.Notification>())).Verifiable();
+            _userRepo.Setup(t => t.GetUserAsync(It.IsAny<string>())).ReturnsAsync(_testUser1).Verifiable();
+        }
+
+        private static Domain.Models.Notification GetDummyNotification(string type)
+        {
+            return new()
+            {
+                Id = "0123",
+                Type = new(type, 1),
+                LoggedInUserId = "test@test.com",
+                TriggerUserId = "test@test.com",
+                HasBeenRead = false,
+                Date = new DateTime(2021, 04, 05)
+            };
+        }
+
+        [Fact]
+        public async Task CommentNotificationAsync_ReturnsOk()
+        {
+            // arrange
+            var dummy = GetDummyNotification("comment");
+
+            NotificationController controller = new(_mockHubContext.Object, _noteRepo.Object, _userRepo.Object);
+
+            var postId = 0;
+
+            // act
+            var result = await controller.CommentNotificationAsync(dummy.LoggedInUserId, dummy.TriggerUserId, postId);
+
+            // assert
+            Assert.NotNull(result);
+            _noteRepo.Verify(c => c.CreateNotificationAsync(It.IsAny<Domain.Models.Notification>()), Times.Once);
+            _userRepo.Verify(c => c.GetUserAsync(It.IsAny<string>()), Times.Once);
+            _mockGroups.Verify(c => c.AddToGroupAsync(It.IsAny<string>(), It.IsAny<string>(), new CancellationToken()), Times.Exactly(3));
+            Assert.IsAssignableFrom<IActionResult>(result);
+            Assert.IsType<OkResult>(result);
```

Comment:
these three assertions on result are redundant in a few ways. the compiler can already guarantee that it's an iactionresult so that call does nothing. and the final IsType call makes the IsAssignableFrom and NotNull calls redundant too.
also, to follow liskov substitution principle, don't require the exact type. so assert `Assert.IsAssignableFrom<OkResult>(result);`, and it replaces the other assertions too. likewise for the similar tests.